### PR TITLE
[TE] return possible values for status, metric, dataset and dimensions in …

### PR DESCRIPTION
**Pull request by kishoreg**
…the anomaly search results.

will update the description with expected output format.

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/pojo/SearchFilters.java [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+import com.linkedin.thirdeye.anomalydetection.context.AnomalyFeedback;
+import com.linkedin.thirdeye.api.DimensionMap;
+import com.linkedin.thirdeye.datalayer.dto.MergedAnomalyResultDTO;
+
+public class SearchFilters {
+
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+  private static final Logger LOG = LoggerFactory.getLogger(SearchFilters.class);
+
+  TreeMap<String, Integer> statusFilterMap;
```
Comment (apucher) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/pojo/SearchFilters.java]: Is there a specific reason to type for on "TreeMap" rather than just "Map"?

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/pojo/SearchFilters.java [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+      if (dimensionFilterMap != null) {
+        DimensionMap dimensions = mergedAnomalyResultDTO.getDimensions();
+        for (String dimensionName : dimensions.keySet()) {
+          if (dimensionFilterMap.containsKey(dimensionName)) {
+            String dimensionValue = dimensions.get(dimensionName);
+            passed = passed && checkFilter(dimensionFilterMap.get(dimensionName), dimensionValue);
+          }
+        }
+      }
+      if (passed) {
```
Comment (cyenjung) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/pojo/SearchFilters.java]: Should we make the check fails fast? When an anomaly does not pass a filter, we skip to the next anomaly immediately.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/pojo/SearchFilters.java [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+import com.linkedin.thirdeye.anomalydetection.context.AnomalyFeedback;
+import com.linkedin.thirdeye.api.DimensionMap;
+import com.linkedin.thirdeye.datalayer.dto.MergedAnomalyResultDTO;
+
+public class SearchFilters {
+
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+  private static final Logger LOG = LoggerFactory.getLogger(SearchFilters.class);
+
+  Map<String, Integer> statusFilterMap;
```
Comment (cyenjung) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/pojo/SearchFilters.java]: It would be helpful to indicate that the value (integer) is the count of appearances of keys.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/pojo/SearchFilters.java [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+    }
+  }
+
+  public static <K, V extends Comparable<? super V>> Map<K, V> sortByValue(Map<K, V> map) {
+    return sortByValue(map, false);
+  }
+
+  public static <K, V extends Comparable<? super V>> Map<K, V> sortByValue(Map<K, V> map, final boolean descending) {
+    final int sortMultipler = (descending) ? -1 : 1;
+    List<Map.Entry<K, V>> list = new LinkedList<Map.Entry<K, V>>(map.entrySet());
```
Comment (cyenjung) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/pojo/SearchFilters.java]: We could use arraylist here since we are not going to insert or remove elements in/from the middle of the list.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/pojo/SearchFilters.java [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+      if (searchFiltersJSON != null) {
+        return OBJECT_MAPPER.readValue(searchFiltersJSON, SearchFilters.class);
+      }
+    } catch (Exception e) {
+      LOG.warn("Exception while parsing searchFiltersJSON  '%s'", searchFiltersJSON, e);
+    }
+    return null;
+  }
+
+  // @Override
```
Comment (cyenjung) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/pojo/SearchFilters.java]: Deadcode :-)

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/pojo/SearchFilters.java (hunk 2 above)
Comment (kishoreg) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/pojo/SearchFilters.java]: I did not want to have too many if statements in the code. once passed becomes false, the remainder of the AND expression should not get invoked. There should be no significant performance penalty

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/pojo/SearchFilters.java (hunk 5 above)
Comment (kishoreg) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/pojo/SearchFilters.java]: I missed this comment, will make this change in another PR
